cg/store/api/status.py

```diff
@@ -553,26 +545,24 @@ def _all_samples_have_sequence_data(links: List[models.FamilySample]) -> bool:
     def analyses_to_upload(self, pipeline: Pipeline = None) -> List[models.Analysis]:
         """Fetch analyses that have not been uploaded"""
         records = self.Analysis.query.join(models.Analysis.family)
-        records = records.filter(
-            models.Analysis.completed_at.isnot(None),
-            models.Analysis.uploaded_at.is_(None),
-            or_(
-                ~models.Family.data_delivery.contains(DataDelivery.SCOUT),
-                and_(
-                    models.Family.data_delivery.contains(DataDelivery.SCOUT),
-                    models.Analysis.delivery_report_created_at.isnot(None),
-                ),
-            ),
-        )
 
-        if pipeline:
-            records = records.filter(models.Analysis.pipeline == str(pipeline))
-            records = (
-                records.filter(models.Family.data_delivery.contains(DataDelivery.SCOUT))
-                if Pipeline.BALSAMIC in pipeline
-                else records
+        analysis_filter_functions: List[str] = [
```

Comment:
🌟 ! So readable and I do not have to figure out any logic. I can just read what is supposed to be filtered. Great!